### PR TITLE
fix(chain_loader): Make getChainInfo async to get chain info after lo…

### PR DESCRIPTION
**Pull request by noahsilencelab**
…ading the file only

### CODE REVIEW

File: lib/screens/wallet/wallet_screen.dart [hunk 1]
```diff
@@ -301,7 +301,7 @@ class _SignScreenState extends State<SignScreen> with WidgetsBindingObserver {
     final analyticManager = Provider.of<AnalyticManager>(context, listen: false);
     final chainLoader = Provider.of<ChainLoader>(context, listen: false);
     _signRequestsSubscription?.pause();
-    Chain? chain = requst.chainId == null ? null : chainLoader.getChainInfo(requst.chainId!);
```
Comment (Granfalloner) [inline lib/screens/wallet/wallet_screen.dart:304]: @noahsilencelab probably there is no reason for us to block UI thread completely and wait for parsing result. 
What do you think about using Future<Chain> in `SingRequestViewModel`, and utilize `FutureBuilder` in UI?

Comment (noahsilencelab) [reply lib/screens/wallet/wallet_screen.dart:304]: @Granfalloner it is a good idea. See implementation: [#ffeabad](https://github.com/silence-laboratories/silent-shard-mobile/pull/19/commits/ffeabadeeec87b7c40d7efc0cf23769c8b8257d8)

File: lib/services/chain_loader.dart [hunk 2]
```diff
@@ -7,15 +7,15 @@ class ChainLoader {
 
   ChainLoader();
 
-  Chain? getChainInfo(int chainId) {
+  Future<Chain> getChainInfo(int? chainId) async {
     if (_chainMap.isEmpty) {
-      _loadChainsAndGenerate();
-      return _chainMap[chainId];
+      await _loadChainsAndGenerate();
+      return Future.value(_chainMap[chainId]);
```
Comment (Granfalloner) [inline lib/services/chain_loader.dart]: This line unconditionally expect chain object and will crash if we don't have particular chain in our database.

Probably we can create some default "placeholder" chain object, which will have real `chainId` and `Unknown` name - in that case, we won't have to handle `null`s.

File: lib/screens/sign/sign_request_view_model.dart [hunk 3]
```diff
@@ -24,9 +24,9 @@ class SignRequestViewModel {
 
   final SignRequest signRequest;
 
-  late final Chain? _chain;
+  late final Future<Chain>? _chain;
```
Comment (Granfalloner) [inline lib/screens/sign/sign_request_view_model.dart]: Looks like we can have non-optional variable here.

File: lib/screens/sign/transaction_details_widget.dart [hunk 4]
```diff
@@ -32,207 +33,224 @@ class TransactionDetailsWidget extends StatelessWidget {
...
-                style: textTheme.bodyMedium,
+      child: FutureBuilder<Chain>(
+        future: requestModel.chain,
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.waiting) {
+            return Center(
+              child: Lottie.asset('assets/lottie/MobileLoader.json'),
+            );
+          }
+          if (snapshot.hasError) {
```
Comment (Granfalloner) [inline lib/screens/sign/transaction_details_widget.dart]: If we failed to load chain info - it probably doesn't mean that we should abort transaction signing. As its not the crucial info - we can still show to user UI without details about chain (besides chain id).

**Review by Granfalloner — APPROVED** [review]

**Pull request merged**